lib/Interpreter/CppInterOp.cpp

```diff
@@ -1657,6 +1657,25 @@ namespace Cpp {
     return QT.getCanonicalType().getAsOpaquePtr();
   }
 
+  bool IsRestrictQualifiedType(TCppType_t type) {
+    if (!type)
+      return 0;
+    QualType QT = QualType::getFromOpaquePtr(type);
+    return QT.isRestrictQualified();
+  }
+
+  TCppType_t GetNonRestrictQualifiedType(TCppType_t type) {
+    if (!type)
+      return 0;
+    QualType QT = QualType::getFromOpaquePtr(type);
+    if (QT.isRestrictQualified()) {
+      QualType NonRestrictType(QT);
+      NonRestrictType.removeLocalRestrict();
+      return NonRestrictType.getAsOpaquePtr();
+    }
+    return nullptr;
```

Comment:
Should we not just return the type as-is if it is not restrict qualified?
```suggestion
    return type;
```

Reply:
Otherwise, LGTM!